cranelift-codegen/meta/src/gen_legalizer.rs

```diff
@@ -134,6 +134,19 @@ fn unwrap_inst(
                     .get(var_pool.get(def.defined_vars[0]).dst_def.unwrap())
                     .to_comment_string(var_pool)
             ));
+
+            fmt.line("let r = pos.func.dfg.inst_results(inst);");
+            for (i, &var_index) in def.defined_vars.iter().enumerate() {
+                let var = var_pool.get(var_index);
+                fmtln!(fmt, "let {} = &r[{}];", var.name, i);
```

Comment:
Is this necessary given that https://github.com/CraneStation/cranelift/pull/833/files#diff-a685da9c9934e511050d2fefc936f786R54 already gets vars.

Reply:
Another one for @bnjbvr

Reply:
@bjorn3 these are different things. The variables defined here are used to be able to reason on the types of result variables. The else branch did generate these result variables and corresponding typeof variables, but the `if` branch didn't, for no good reasons. (It also landed in a separate PR, see [commit](https://github.com/CraneStation/cranelift/commit/6a2a5e4dde73c16256c484252e77d804cfc55676)).